tests/evhtests/l7_evh_crd_test.go

```diff
@@ -722,23 +722,97 @@ func TestHostruleAnalyticsPolicyUpdateForEvh(t *testing.T) {
 	g.Expect(nodes[0].EvhNodes).To(gomega.HaveLen(1))
 	g.Expect(nodes[0].EvhNodes[0].AnalyticsPolicy).To(gomega.BeNil())
 
-	// Update host rule with AnalyticsPolicy
+	// Update host rule with AnalyticsPolicy - only with LogallHeaders
 	hrUpdate := integrationtest.FakeHostRule{
 		Name:      hrName,
 		Namespace: "default",
 		Fqdn:      "foo.com",
 	}.HostRule()
 	enabled := true
 	analyticsPolicy := &v1beta1.HostRuleAnalyticsPolicy{
+		LogAllHeaders: &enabled,
+	}
+	hrUpdate.Spec.VirtualHost.AnalyticsPolicy = analyticsPolicy
+	hrUpdate.ResourceVersion = "2"
+	_, err := v1beta1CRDClient.AkoV1beta1().HostRules("default").Update(context.TODO(), hrUpdate, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating HostRule: %v", err)
+	}
+	g.Eventually(func() string {
+		hostrule, _ := v1beta1CRDClient.AkoV1beta1().HostRules("default").Get(context.TODO(), hrName, metav1.GetOptions{})
+		return hostrule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+
+	g.Eventually(func() int {
+		_, aviModel = objects.SharedAviGraphLister().Get(modelName)
+		nodes = aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		return len(nodes[0].EvhNodes)
+	}, 10*time.Second).Should(gomega.Equal(1))
+
+	// update is not getting reflected on evh nodes immediately. Hence adding a sleep of 5 seconds.
+	time.Sleep(5 * time.Second)
+
+	// Check whether the AnalyticsPolicy values are properly updated
+	g.Expect(nodes[0].EvhNodes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].EvhNodes[0].AnalyticsPolicy).ShouldNot(gomega.BeNil())
+	g.Expect(*nodes[0].EvhNodes[0].AnalyticsPolicy.AllHeaders).To(gomega.BeTrue())
+
+	// Update host rule with AnalyticsPolicy - with LogAllHeader, Enabled
```

Comment:
We can update this to :
Update host rule with AnalyticsPolicy - with LogAllHeader, FullClientLogs.Enabled